modules/gitops-argo-rollouts-architecture-overview.adoc

```diff
@@ -0,0 +1,70 @@
+|===
+
+[id="gitops-components_{context}"]
+== Components
+
+Argo Rollouts consists of several components that enable users to practice Progressive Delivery on OpenShift.
+
+[options="header"]
+|===
+|Name | Description
+|`Rollouts Controller` | The Rollouts Controller component of the Argo Rollouts manages the Rollouts custom resource which resembles a native Kubernetes Deployment resource. The Rollouts Controller does not modify the standard deployment resources and only responds to changes in the Rollouts resources.
+|`AnalysisRun Controller` | The `AnalysisRun` controller performs analysis for an `AnalysisRun` resource and manages the `AnalysisRun` custom resource. It provides the capability to connect a Rollout to your metrics provider and define specific thresholds for certain metrics that decide if a deployment update is successful for your application.
+
+Argo Rollouts supports two custom Kubernetes resources to perform Analysis - `AnalysisTemplate` and `AnalysisRun`. You can define an `AnalysisTemplate` on a specific Rollout or globally on a cluster to be shared by multiple rollouts as a Cluster`AnalysisTemplate`. The `AnalysisRun` resource is scoped on a specific rollout.
+|`Experiment Controller` | The `Experiment` Controller creates analysis for limited-duration, short-lived replica sets, and manages the `Experiment` custom resource. The `Experiment` custom resource can be used independently or executed as part of Rollouts. You can link a list of `AnalysisTemplate` to be used on replica sets by creating an `Experiment` custom resource.
+
+In addition, this controller is used for creating replica sets. You can use it for optionally creating the service, running the analysis, reporting the results, and scaling down the replica sets in the end. The `Experiment` controller can also be integrated with Rollouts by specifying the experiment step in the canary deployment strategy.
+|`Service and Ingress Controller` | The `Service` controller handles the `Service` resources and the `Ingress` controller manages the `Ingress` resources modified by Argo Rollouts. These controllers inject additional metadata annotations in the application instances for traffic management.
+
+Argo Rollouts supports the standard Kubernetes Service and Ingress resources for forwarding the incoming user traffic to the correct version of an application. Depending on the deployment strategy and the traffic management tool used by your application, you can configure Argo Rollouts to split incoming traffic in application instances with specific percentages.
+|`UI and CLI` | Argo Rollouts supports an `oc/kubectl` plugin to interact with resources like Rollouts, Analysis, and Experiments from the command line. It can perform operations like pause, promote, retry, on the Rollout resource. The Argo Rollouts CLI plugin can start a local web UI dashboard in the browser to enhance the experience of visualizing the Rollouts resources.
```

Comment:
```suggestion
|Argo Rollouts CLI and UI |Argo Rollouts supports an `oc/kubectl` plugin called Argo Rollouts CLI . You can use it to interact with resources , such as rollouts, analyses, and experiments, from the command line. It can perform operations, such as `pause`, `promote`, or `retry`. The Argo Rollouts CLI plugin can start a local web UI dashboard in the browser to enhance the experience of visualizing the Argo Rollouts resources.
```
I had the similar question in the Argo CLI PR: should we refer to it as `oc/kubectl` plugin downstream?

Reply:
Fixed.

Reply:
@jgwest - FYI

Reply:
It is a plugin for the `oc` and `kubectl` CLIs, using the standard plugin mechanism of those CLIs (`oc` is based on `kubectl`). It integrates into the existing `oc` / `kubectl` CLI, allowing you to run Rollout commands just by calling `oc` or `kubectl`. Not sure what you all want to refer to it as, but this is its function.

Reply:
Great, thank you @jgwest for an explanation. I was not sure if the CLI would run `kubectl` commands or just `oc`. If it is capable of running both, then it should be okay to have it called a `oc/kubectl`  plugin in our docs :+1:

Reply:
Keeping it as kubetcl / oc CLI plugin